src/simtools/data_model/metadata_collector.py

```diff
@@ -256,31 +335,32 @@ def _read_input_metadata_from_file(self, metadata_file_name=None):
             if metadata does not exist
 
         """
-        metadata_file_name = (
-            self.args_dict.get("input_meta", None) or self.args_dict.get("input", None)
-            if metadata_file_name is None
-            else metadata_file_name
+        metadata_file_names = (
+            metadata_file_name_expression
+            or self.args_dict.get("input_meta")
+            or self.args_dict.get("input")
         )
 
-        if metadata_file_name is None:
+        try:
+            metadata_files = gen.generate_list_of_files(metadata_file_names)
+        except ValueError:
             self._logger.debug("No input metadata file defined.")
-            return {}
-
-        self._logger.debug("Reading meta data from %s", metadata_file_name)
-        if Path(metadata_file_name).suffix in (".yaml", ".yml", ".json"):
-            _input_metadata = self._read_input_metadata_from_yml_or_json(metadata_file_name)
-        elif Path(metadata_file_name).suffix == ".ecsv":
-            _input_metadata = self._read_input_metadata_from_ecsv(metadata_file_name)
-        else:
-            self._logger.error("Unknown metadata file format: %s", metadata_file_name)
-            raise gen.InvalidConfigDataError
+            return None
+
+        metadata = []
+        for metadata_file in metadata_files:
+            self._logger.debug("Reading meta data from %s", metadata_file)
```

Comment:
That's the correct way to do string formatting in logging, but I thought we were consistent in not using it anywhere and choose f-string instead. Not a big deal either way.

Reply:
Fixed.